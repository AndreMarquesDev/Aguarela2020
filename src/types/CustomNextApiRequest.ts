import type { NextApiRequest } from 'next';
import type { FormPostRequestBody } from './FormPostRequestBody';

export type CustomNextApiRequest = {
    body: FormPostRequestBody;
} & NextApiRequest;
