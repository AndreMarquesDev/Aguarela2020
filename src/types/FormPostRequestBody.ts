import type { FormValues } from './FormValues';

export type FormPostRequestBody = {
    isPlaywright: boolean;
} & FormValues;
