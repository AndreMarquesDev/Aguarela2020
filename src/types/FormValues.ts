export type FormValues = {
    name: string;
    email: string;
    message: string;
    brand: string;
    subject: string;
};
