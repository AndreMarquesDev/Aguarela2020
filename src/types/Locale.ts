export enum Locale {
    Pt = 'pt',
    En = 'en',
}
