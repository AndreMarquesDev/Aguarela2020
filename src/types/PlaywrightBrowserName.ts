export enum PlaywrightBrowserName {
    Chromium = 'chromium',
    Firefox = 'firefox',
    Webkit = 'webkit',
}
