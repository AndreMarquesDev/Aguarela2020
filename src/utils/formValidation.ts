export enum FieldTypes {
    Name = 'name',
    Email = 'email',
    Textarea = 'message',
    Brand = 'brand',
    Subject = 'subject',
}
export type FormState = {
    name: string;
    email: string;
    message: string;
    brand: string;
    subject: string;
};

export const validateName = (value: string): boolean => {
    const nameRegex =
        // eslint-disable-next-line regexp/no-unused-capturing-group, regexp/prefer-plus-quantifier, regexp/prefer-character-class, regexp/no-super-linear-backtracking, regexp/no-useless-escape
        /s*([a-zA-ZáâàãéêèíîìóôòõúûùñçäöüßÁÂÀÃÉÊÈÍÎÌÓÔÒÕÚÛÙÑÇÄÖÜẞ]{1,}([\.,] |[-']| ))+[a-zA-ZáâàãéêèíîìóôòõúûùñçäöüßÁÂÀÃÉÊÈÍÎÌÓÔÒÕÚÛÙÑÇÄÖÜẞ]+\.?\s*$/;

    return nameRegex.test(value);
};

export const validateEmail = (value: string): boolean => {
    const emailRegex =
        // eslint-disable-next-line regexp/no-useless-escape, regexp/no-unused-capturing-group, regexp/prefer-d, regexp/use-ignore-case
        /(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;

    return emailRegex.test(value);
};
