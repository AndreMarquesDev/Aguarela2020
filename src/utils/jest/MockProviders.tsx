/* eslint-disable react/jsx-no-constructed-context-values */
import React, { ReactNode } from 'react';
import { NavLinksContext } from '../../components/context/NavLinksContext';
import { TextsContext } from '../../components/context/TextsContext';
import { Locale } from '../../types/Locale';
import { textsEn, textsPt } from '../texts';

interface MockProvidersProps {
    language?: Locale;
    isMenuOpen?: boolean;
    toggleMenu?: jest.Mock;
    setNavHeight?: jest.Mock;
    children: ReactNode;
}

export const MockProviders = ({
    language = Locale.Pt,
    isMenuOpen = false,
    toggleMenu = jest.fn(),
    setNavHeight = undefined,
    children,
}: MockProvidersProps): JSX.Element => {
    return (
        <TextsContext.Provider
            value={{
                texts: language === Locale.Pt ? textsPt : textsEn,
            }}
        >
            <NavLinksContext.Provider value={{ isMenuOpen, toggleMenu, setNavHeight }}>
                {children}
            </NavLinksContext.Provider>
        </TextsContext.Provider>
    );
};
