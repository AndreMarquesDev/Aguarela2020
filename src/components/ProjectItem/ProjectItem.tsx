import React, { useContext, TouchEvent } from 'react';
import Image from 'next/image';
import { isClientSide } from 'multilingual-url/lib';
import classNames from 'classnames';
import { TextsContext } from '../context/TextsContext';
import { resetTimesTouchedAttribute } from '../../utils/generic';
import { projectItemTouchDivDataTestId } from '../../utils/dataTestIds';
import { desktopMediaQuery, phabletMediaQuery, tabletMediaQuery } from '../../styles/mediaQueries';

export interface ProjectItemProps {
    imageSrc: string;
    imageAlt: string;
    brandLink: string;
    brandTag: string;
    description: string;
    isDesktop?: boolean;
    year?: number | string;
    isInPartnership?: boolean;
    isActive?: boolean;
    isGrid?: boolean;
    preloadImage?: boolean;
}

export const ProjectItem = ({
    imageSrc,
    imageAlt,
    brandLink,
    brandTag,
    description,
    year,
    isDesktop,
    isInPartnership,
    isActive,
    isGrid,
    preloadImage,
}: ProjectItemProps): JSX.Element => {
    const { texts } = useContext(TextsContext);

    const imageSize = 400;
    const isTouch = isClientSide && window?.matchMedia('(hover: none), (pointer: coarse)').matches;

    const handleTouch = (event: TouchEvent<HTMLElement>): void => {
        const target = event.currentTarget;
        const timesTouched = parseInt(target.dataset.timesTouched as string);

        if (timesTouched === 0) {
            resetTimesTouchedAttribute();
            target.dataset.timesTouched = '1';
        } else {
            (target.querySelector('a.brand.link') as HTMLAnchorElement).click();
        }
    };

    return (
        <>
            {isDesktop ? (
                <li>
                    <Image
                        alt={imageAlt}
                        height={imageSize}
                        priority={!!preloadImage}
                        src={imageSrc}
                        style={{
                            width: '100%',
                            height: 'auto',
                            position: 'relative',
                            maxWidth: `${imageSize}px`,
                        }}
                        width={imageSize}
                    />
                    <div className="backface">
                        <a
                            className="brand link fontXS"
                            href={brandLink}
                            rel="noreferrer"
                            target="_blank"
                        >
                            {brandTag}
                        </a>
                        <p className="description fontS">{description}</p>
                        {year && (
                            <p className="date fontXS">
                                {year}
                                {isActive && ` - ${texts.present}`}
                            </p>
                        )}
                        {isInPartnership && (
                            <small className="fontXS">* {texts.inPartnershipWith}</small>
                        )}
                    </div>
                </li>
            ) : (
                <div
                    className={classNames('carouselItem', isTouch && 'touch', isGrid && 'grid')}
                    data-testid={`${projectItemTouchDivDataTestId}_${imageAlt}`}
                    data-times-touched={0}
                    onTouchStart={handleTouch}
                >
                    <Image
                        alt={imageAlt}
                        height={imageSize}
                        priority={!!preloadImage}
                        src={imageSrc}
                        style={{
                            width: '100%',
                            height: 'auto',
                            position: 'relative',
                            maxWidth: `${imageSize}px`,
                        }}
                        width={imageSize}
                    />
                    <div className="backface">
                        <a
                            className="brand link fontXS"
                            href={brandLink}
                            rel="noreferrer"
                            target="_blank"
                        >
                            {brandTag}
                        </a>
                        <p className="description fontS">{description}</p>
                        {year && (
                            <p className="date fontXS">
                                {year}
                                {isActive && ` - ${texts.present}`}
                            </p>
                        )}
                        {isInPartnership && (
                            <small className="fontXS">* {texts.inPartnershipWith}</small>
                        )}
                    </div>
                </div>
            )}

            <style jsx>
                {`
                    li {
                        max-height: 400rem;
                        position: relative;
                        margin-right: 2%;

                        &:last-child {
                            margin-right: 0;
                        }

                        &:hover {
                            .backface {
                                display: flex;
                            }
                        }
                    }

                    .carouselItem {
                        max-width: 400rem;
                        max-height: 400rem;
                        width: auto;
                        height: auto;
                        display: flex;
                        flex-direction: column;
                        align-items: center;
                        text-align: center;
                        position: relative;
                        margin: 0 auto;

                        &:hover {
                            .backface {
                                display: flex;
                            }
                        }

                        &.grid {
                            margin: 0 4% 0;

                            @media (${tabletMediaQuery}) {
                                margin: 0 2% 0;
                            }

                            .backface {
                                .brand {
                                    color: var(--yellow);
                                    text-transform: uppercase;

                                    @media (${tabletMediaQuery}) {
                                        font-size: var(--fontSizeXXS);
                                    }
                                }

                                .date {
                                    color: var(--yellow);

                                    @media (${tabletMediaQuery}) {
                                        font-size: var(--fontSizeXXS);
                                    }
                                }

                                .description {
                                    color: var(--yellow);

                                    @media (${desktopMediaQuery}) {
                                        font-size: var(--fontSizeXS);
                                    }

                                    @media (${tabletMediaQuery}) {
                                        font-size: var(--fontSizeXXS);
                                    }
                                }

                                small {
                                    color: var(--yellow);

                                    @media (${tabletMediaQuery}) {
                                        font-size: var(--fontSizeXXS);
                                    }
                                }
                            }
                        }
                    }

                    .backface {
                        display: none;
                        width: 100%;
                        height: 100%;
                        position: absolute;
                        top: 0;
                        left: 0;
                        flex-direction: column;
                        justify-content: center;
                        align-items: center;
                        text-align: center;
                        background: var(--blue);
                        padding: 10rem;

                        .brand {
                            color: var(--yellow);
                            text-transform: uppercase;
                            margin-bottom: 20rem;
                        }

                        .date {
                            color: var(--yellow);
                            margin-bottom: 20rem;
                        }

                        .description {
                            color: var(--yellow);
                            margin-bottom: 20rem;

                            @media (${phabletMediaQuery}) {
                                font-size: var(--fontSizeXS);
                            }
                        }

                        small {
                            color: var(--yellow);
                        }
                    }
                `}
            </style>
        </>
    );
};
