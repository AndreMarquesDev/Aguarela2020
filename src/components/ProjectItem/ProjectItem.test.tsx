import type { RenderResult } from '@testing-library/react';
import type { ProjectItemProps } from './ProjectItem';
import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import { Locale } from '../../types/Locale';
import { projectItemTouchDivDataTestId } from '../../utils/dataTestIds';
import { MockProviders } from '../../utils/jest/MockProviders';
import { textsEn, textsPt } from '../../utils/texts';
import { tjelaInstagramUrl } from '../../utils/urls';
import { ProjectItem } from './ProjectItem';
import '@testing-library/jest-dom';

const baseProps: ProjectItemProps = {
    brandLink: tjelaInstagramUrl,
    brandTag: '@tudonatjela',
    description: textsPt.socialMediaManagementAndContentCreation,
    imageAlt: 'tjela logo',
    imageSrc: '/images/tjela.jpg',
    isDesktop: true,
    isInPartnership: true,
    isActive: true,
    isGrid: true,
    year: 2020,
    preloadImage: true,
};

const renderComponent = (
    newProps?: Partial<ProjectItemProps>,
    language: Locale = Locale.Pt,
): RenderResult => {
    return render(
        <MockProviders language={language}>
            <ProjectItem {...baseProps} {...newProps} />
        </MockProviders>,
    );
};

describe('<ProjectItem />', () => {
    it('renders properly', () => {
        const { container } = renderComponent();

        const image = screen.getByAltText(baseProps.imageAlt);
        const brandLink = screen.getByText(baseProps.brandTag);
        const description = screen.getByText(baseProps.description);
        const year = screen.getByText(`${baseProps.year} - ${textsPt.present}`);
        const isInPartership = screen.getByText(`* ${textsPt.inPartnershipWith}`);

        expect(image).toBeInTheDocument();
        expect(brandLink).toBeInTheDocument();
        expect(description).toBeInTheDocument();
        expect(year).toBeInTheDocument();
        expect(isInPartership).toBeInTheDocument();

        expect(brandLink).toHaveAttribute('href', baseProps.brandLink);

        expect(container).toMatchSnapshot();
    });

    it('renders properly in English', () => {
        const { container } = renderComponent(
            { description: textsEn.socialMediaManagementAndContentCreation },
            Locale.En,
        );

        const year = screen.getByText(`${baseProps.year} - ${textsEn.present}`);
        const isInPartership = screen.getByText(`* ${textsEn.inPartnershipWith}`);

        expect(year).toBeInTheDocument();
        expect(isInPartership).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });

    it('renders properly in mobile', () => {
        const { container } = renderComponent({ isDesktop: false });

        const image = screen.getByAltText(baseProps.imageAlt);
        const brandLink = screen.getByText(baseProps.brandTag);
        const description = screen.getByText(baseProps.description);
        const year = screen.getByText(`${baseProps.year} - ${textsPt.present}`);
        const isInPartership = screen.getByText(`* ${textsPt.inPartnershipWith}`);

        expect(image).toBeInTheDocument();
        expect(brandLink).toBeInTheDocument();
        expect(description).toBeInTheDocument();
        expect(year).toBeInTheDocument();
        expect(isInPartership).toBeInTheDocument();

        expect(brandLink).toHaveAttribute('href', baseProps.brandLink);

        expect(container).toMatchSnapshot();
    });

    it('renders properly with the year as a string', () => {
        renderComponent({ year: '2019 - 2020' });

        const year = screen.getByText(`2019 - 2020 - ${textsPt.present}`);

        expect(year).toBeInTheDocument();
    });

    it('renders properly with isActive being false', () => {
        renderComponent({ isActive: false });

        const year = screen.getByText(baseProps.year as number);

        const presentText = screen.queryByText(textsPt.present);

        expect(year).toBeInTheDocument();

        expect(presentText).not.toBeInTheDocument();
    });

    it('renders properly with isInPartnership being false', () => {
        renderComponent({ isInPartnership: false });

        const isInPartership = screen.queryByText(`* ${textsPt.inPartnershipWith}`);

        expect(isInPartership).not.toBeInTheDocument();
    });

    it('renders properly with isGrid being false', () => {
        const { container } = renderComponent({ isGrid: false, isDesktop: false });

        expect(container).toMatchSnapshot();
    });

    it('renders properly with preloadImage being false', () => {
        const { container } = renderComponent({ preloadImage: false });

        expect(container).toMatchSnapshot();
    });

    it('renders properly when device is touch', () => {
        window.matchMedia = jest.fn().mockImplementation(() => ({
            matches: true,
        }));

        const { container } = renderComponent({ isDesktop: false });

        const wrapper = screen.getByTestId(
            `${projectItemTouchDivDataTestId}_${baseProps.imageAlt}`,
        );

        expect(wrapper).toHaveAttribute('data-times-touched', '0');

        fireEvent.touchStart(wrapper);

        expect(wrapper).toHaveAttribute('data-times-touched', '1');

        fireEvent.touchStart(wrapper);

        expect(wrapper).toHaveAttribute('data-times-touched', '1');

        expect(container).toMatchSnapshot();
    });
});
