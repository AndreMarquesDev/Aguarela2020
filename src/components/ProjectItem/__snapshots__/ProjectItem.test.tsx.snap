// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectItem /> renders properly 1`] = `
<div>
  <li
    class="jsx-2553597104"
  >
    <img
      alt="tjela logo"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="400"
      src="/_next/image?url=%2Fimages%2Ftjela.jpg&w=828&q=75"
      srcset="/_next/image?url=%2Fimages%2Ftjela.jpg&w=640&q=75 1x, /_next/image?url=%2Fimages%2Ftjela.jpg&w=828&q=75 2x"
      style="color: transparent; width: 100%; height: auto; position: relative; max-width: 400px;"
      width="400"
    />
    <div
      class="jsx-2553597104 backface"
    >
      <a
        class="jsx-2553597104 brand link fontXS"
        href="https://www.instagram.com/tudonatjela/"
        rel="noreferrer"
        target="_blank"
      >
        @tudonatjela
      </a>
      <p
        class="jsx-2553597104 description fontS"
      >
        Gestão de Redes Sociais e Criação de Conteúdo
      </p>
      <p
        class="jsx-2553597104 date fontXS"
      >
        2020
         - Presente
      </p>
      <small
        class="jsx-2553597104 fontXS"
      >
        *
         
        em dupla com Sofia Ferreirinho
      </small>
    </div>
  </li>
</div>
`;

exports[`<ProjectItem /> renders properly in English 1`] = `
<div>
  <li
    class="jsx-2553597104"
  >
    <img
      alt="tjela logo"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="400"
      src="/_next/image?url=%2Fimages%2Ftjela.jpg&w=828&q=75"
      srcset="/_next/image?url=%2Fimages%2Ftjela.jpg&w=640&q=75 1x, /_next/image?url=%2Fimages%2Ftjela.jpg&w=828&q=75 2x"
      style="color: transparent; width: 100%; height: auto; position: relative; max-width: 400px;"
      width="400"
    />
    <div
      class="jsx-2553597104 backface"
    >
      <a
        class="jsx-2553597104 brand link fontXS"
        href="https://www.instagram.com/tudonatjela/"
        rel="noreferrer"
        target="_blank"
      >
        @tudonatjela
      </a>
      <p
        class="jsx-2553597104 description fontS"
      >
        Social Media Management and Content Creation
      </p>
      <p
        class="jsx-2553597104 date fontXS"
      >
        2020
         - Present
      </p>
      <small
        class="jsx-2553597104 fontXS"
      >
        *
         
        in partnership with Sofia Ferreirinho
      </small>
    </div>
  </li>
</div>
`;

exports[`<ProjectItem /> renders properly in mobile 1`] = `
<div>
  <div
    class="jsx-2553597104 carouselItem grid"
    data-testid="projectItem_touch_div_tjela logo"
    data-times-touched="0"
  >
    <img
      alt="tjela logo"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="400"
      sizes="(max-width: 1023px) 50vw, (max-width: 767px) 75vw, 30vw"
      src="/_next/image?url=%2Fimages%2Ftjela.jpg&w=3840&q=75"
      srcset="/_next/image?url=%2Fimages%2Ftjela.jpg&w=256&q=75 256w, /_next/image?url=%2Fimages%2Ftjela.jpg&w=384&q=75 384w, /_next/image?url=%2Fimages%2Ftjela.jpg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Ftjela.jpg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Ftjela.jpg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Ftjela.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Ftjela.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Ftjela.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Ftjela.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Ftjela.jpg&w=3840&q=75 3840w"
      style="color: transparent; width: 100%; height: auto; position: relative; max-width: 400px;"
      width="400"
    />
    <div
      class="jsx-2553597104 backface"
    >
      <a
        class="jsx-2553597104 brand link fontXS"
        href="https://www.instagram.com/tudonatjela/"
        rel="noreferrer"
        target="_blank"
      >
        @tudonatjela
      </a>
      <p
        class="jsx-2553597104 description fontS"
      >
        Gestão de Redes Sociais e Criação de Conteúdo
      </p>
      <p
        class="jsx-2553597104 date fontXS"
      >
        2020
         - Presente
      </p>
      <small
        class="jsx-2553597104 fontXS"
      >
        *
         
        em dupla com Sofia Ferreirinho
      </small>
    </div>
  </div>
</div>
`;

exports[`<ProjectItem /> renders properly when device is touch 1`] = `
<div>
  <div
    class="jsx-2553597104 carouselItem touch grid"
    data-testid="projectItem_touch_div_tjela logo"
    data-times-touched="1"
  >
    <img
      alt="tjela logo"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="400"
      sizes="(max-width: 1023px) 50vw, (max-width: 767px) 75vw, 30vw"
      src="/_next/image?url=%2Fimages%2Ftjela.jpg&w=3840&q=75"
      srcset="/_next/image?url=%2Fimages%2Ftjela.jpg&w=256&q=75 256w, /_next/image?url=%2Fimages%2Ftjela.jpg&w=384&q=75 384w, /_next/image?url=%2Fimages%2Ftjela.jpg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Ftjela.jpg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Ftjela.jpg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Ftjela.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Ftjela.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Ftjela.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Ftjela.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Ftjela.jpg&w=3840&q=75 3840w"
      style="color: transparent; width: 100%; height: auto; position: relative; max-width: 400px;"
      width="400"
    />
    <div
      class="jsx-2553597104 backface"
    >
      <a
        class="jsx-2553597104 brand link fontXS"
        href="https://www.instagram.com/tudonatjela/"
        rel="noreferrer"
        target="_blank"
      >
        @tudonatjela
      </a>
      <p
        class="jsx-2553597104 description fontS"
      >
        Gestão de Redes Sociais e Criação de Conteúdo
      </p>
      <p
        class="jsx-2553597104 date fontXS"
      >
        2020
         - Presente
      </p>
      <small
        class="jsx-2553597104 fontXS"
      >
        *
         
        em dupla com Sofia Ferreirinho
      </small>
    </div>
  </div>
</div>
`;

exports[`<ProjectItem /> renders properly with isGrid being false 1`] = `
<div>
  <div
    class="jsx-2553597104 carouselItem"
    data-testid="projectItem_touch_div_tjela logo"
    data-times-touched="0"
  >
    <img
      alt="tjela logo"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="400"
      sizes="(max-width: 1023px) 50vw, (max-width: 767px) 75vw, 30vw"
      src="/_next/image?url=%2Fimages%2Ftjela.jpg&w=3840&q=75"
      srcset="/_next/image?url=%2Fimages%2Ftjela.jpg&w=256&q=75 256w, /_next/image?url=%2Fimages%2Ftjela.jpg&w=384&q=75 384w, /_next/image?url=%2Fimages%2Ftjela.jpg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Ftjela.jpg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Ftjela.jpg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Ftjela.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Ftjela.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Ftjela.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Ftjela.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Ftjela.jpg&w=3840&q=75 3840w"
      style="color: transparent; width: 100%; height: auto; position: relative; max-width: 400px;"
      width="400"
    />
    <div
      class="jsx-2553597104 backface"
    >
      <a
        class="jsx-2553597104 brand link fontXS"
        href="https://www.instagram.com/tudonatjela/"
        rel="noreferrer"
        target="_blank"
      >
        @tudonatjela
      </a>
      <p
        class="jsx-2553597104 description fontS"
      >
        Gestão de Redes Sociais e Criação de Conteúdo
      </p>
      <p
        class="jsx-2553597104 date fontXS"
      >
        2020
         - Presente
      </p>
      <small
        class="jsx-2553597104 fontXS"
      >
        *
         
        em dupla com Sofia Ferreirinho
      </small>
    </div>
  </div>
</div>
`;

exports[`<ProjectItem /> renders properly with preloadImage being false 1`] = `
<div>
  <li
    class="jsx-2553597104"
  >
    <img
      alt="tjela logo"
      data-nimg="1"
      decoding="async"
      height="400"
      loading="lazy"
      src="/_next/image?url=%2Fimages%2Ftjela.jpg&w=828&q=75"
      srcset="/_next/image?url=%2Fimages%2Ftjela.jpg&w=640&q=75 1x, /_next/image?url=%2Fimages%2Ftjela.jpg&w=828&q=75 2x"
      style="color: transparent; width: 100%; height: auto; position: relative; max-width: 400px;"
      width="400"
    />
    <div
      class="jsx-2553597104 backface"
    >
      <a
        class="jsx-2553597104 brand link fontXS"
        href="https://www.instagram.com/tudonatjela/"
        rel="noreferrer"
        target="_blank"
      >
        @tudonatjela
      </a>
      <p
        class="jsx-2553597104 description fontS"
      >
        Gestão de Redes Sociais e Criação de Conteúdo
      </p>
      <p
        class="jsx-2553597104 date fontXS"
      >
        2020
         - Presente
      </p>
      <small
        class="jsx-2553597104 fontXS"
      >
        *
         
        em dupla com Sofia Ferreirinho
      </small>
    </div>
  </li>
</div>
`;
